recipe/ingredients.py

```diff
@@ -503,6 +515,21 @@ def __init__(self, expression, **kwargs):
             else:
                 self.formatters.insert(0, lambda value: self.lookup.get(value, value))
 
+    def get_group_by(self):
+        # Ensure the labels are generated
+        if not self._labels:
+            list(self.query_columns)
+
+        if self.group_by_strategy == "labels":
+            return [lbl for gb, lbl in zip(self._group_by, self._labels)]
+        else:
+            return self._group_by
+
+    def set_group_by(self, value):
+        self._group_by = value
+
+    group_by = property(get_group_by, set_group_by)
```

Comment:
You can use a slightly more ergonomic syntax for doing property setters:

```
@property
def group_by(self):
    ...

@group_by.setter
def group_by(self, value):
    self._group_by = value
```

Reply:
Thanks, bikeshedding is :+1: